File: library-ui/src/app/interfaces/admin-book-dto.ts
export interface AdminBookDto {
  id?: number;
  categoryId?: number;
  publishYear?: number;
  pages?: number;
  coverImageUrl?: string;
  authorIds?: number[];
  translations: { [key: string]: AdminBookTranslationDto };
}

export interface AdminBookTranslationDto {
  title: string;
  bookLanguage: string;
  description: string;
}

File: library-app/src/main/java/org/example/library/admin/dto/AdminBookDto.java
package org.example.library.admin.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.Map;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class AdminBookDto {
    private Integer id;
    private Integer categoryId;
    private Short publishYear;
    private Short pages;
    private String coverImageUrl;
    private Map<String, AdminBookTranslationDto> translations;
    private Iterable<Integer> authorIds;

    @Getter
    @Setter
    @NoArgsConstructor
    @AllArgsConstructor
    @Builder
    public static class AdminBookTranslationDto {
        private String title;
        private String bookLanguage;
        private String description;
    }
}
